src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -280,10 +280,26 @@ private bool IsValidSlnFilePath(string path)
 
             string[] globalPropEnumerable = _parseResult.GetValue(CommonOptions.PropertiesOption);
 
-            foreach (var keyEqVal in globalPropEnumerable)
+            foreach (var keyEqValString in globalPropEnumerable)
             {
-                string[] keyValuePair = keyEqVal.Split("=", 2);
-                globalProperties[keyValuePair[0]] = keyValuePair[1];
+                foreach (var keyEqVal in keyEqValString.Split(";"))
+                {
+                    string[] keyValuePair = keyEqVal.Split("=", 2);
+                    if (keyValuePair.Count() == 0)
+                    {
+                        // The user provided an empty value after ;. e.g. -p:Foo=Bar;
+                        continue;
+                    }
+                    else if (keyValuePair.Count() == 1)
+                    {
+                        // The user provided a property such as -p:Foo without giving a value. Undefine the property.
+                        // Note that MSBuild will fail if given -p:Foo; but we can ignore this situation because of this and let MsBuild fail, or, assuming it comes to parity with ';', behave the same as if ; was not present.
+                        // The same can be said for -p:Foo=bar;-p:Bar=Foo, which is also invalid syntax to MSBuild.
+                        globalProperties[keyValuePair[0]] = "";
```

Comment:
Do you need to specifically set this to empty, or can you just skip adding the key to the dictionary?

Reply:
We should. If we skip adding it to the dictionary, it will not mimic Msbuild behavior... for example, if I do -p:_IsPublishing, that will globally undefine, permanently, the _IsPublishing property ... if we dont do the same for these properties we inject into the pre-evaluation, the pre-evaluation can yield inconsistent results with what msbuild would actually do

Reply:
A lot of this parsing seems to overlap with [MSBuildPropertyParser](https://teams.microsoft.com/l/channel/19%3a2db9948434504972b9bc70d3b3ad71e2%40thread.skype/CLI?groupId=4ba7372f-2799-4677-89f0-7a1aaea3706c&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47), which currently is only used when calculating the values to 'forward' to MSBuild. Is there opportunity for reuse here, or potentially even changing the type of the `PropertiesOption` away from `string[]` to a type that more accurately describes the semantics?

Reply:
We should re-use it for sure, initially we used it. I thought it handled the `-p` `--p` `/p` shenanigans for some reason when I initially used it, and when I realized I didn't I went back to using the PropertiesOption because I thought it handled everything. Obviously, my world view was incorrect. 

BTW, that link is to our teams channel, should change and delete the edit history ;)

Reply:
Using it again now 🎉